headless-server-commerce-base-lib/src/main/java/com/coremedia/blueprint/headlessserver/CommerceCaasConfig.java

```diff
@@ -0,0 +1,410 @@
+import static com.coremedia.caas.headless_server.plugin_support.PluginSupport.QUALIFIER_PLUGIN_WIRING_FACTORIES;
+import static java.lang.invoke.MethodHandles.lookup;
+import static java.util.Collections.emptyList;
+
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties({
+        CaasServiceConfigurationProperties.class,
+        CaasGraphqlConfigurationProperties.class,
+        RemoteServiceConfigurationProperties.class,
+        StaxContextConfigurationProperties.class,
+        ContentConfigurationProperties.class,
+        DoctypeConfigurationProperties.class,
+})
+@EnableWebMvc
+@Import({
+        // inspired by CaasConfig
+        CacheConfiguration.class,
+        CustomizerConfiguration.class,
+        CapRepositoriesConfiguration.class,
+        GraphQlConfiguration.class,
+        PluginConfiguration.class,
+        CaasWebConfig.class,
+        GraphQLErrorController.class,
+})
+
+/**
+ * Copy and adapt the configuration from the CaasConfig class.
+ */
+public class CommerceCaasConfig implements WebMvcConfigurer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(lookup().lookupClass());
+  private final CaasServiceConfigurationProperties caasServiceConfigurationProperties;
+  private final CaasGraphqlConfigurationProperties caasGraphqlConfigurationProperties;
+
+  public CommerceCaasConfig(CaasServiceConfigurationProperties caasServiceConfigurationProperties, CaasGraphqlConfigurationProperties caasGraphqlConfigurationProperties) {
+    this.caasServiceConfigurationProperties = caasServiceConfigurationProperties;
+    this.caasGraphqlConfigurationProperties = caasGraphqlConfigurationProperties;
+  }
+
+  @Bean
+  @Qualifier("graphqlSchemaResource")
+  public Resource commerceBaseSchemaResource() throws IOException {
+    PathMatchingResourcePatternResolver loader = new PathMatchingResourcePatternResolver();
+    return Arrays.stream(loader.getResources("classpath*:commerce-base-schema.graphql"))
+            .findFirst()
+            .orElseThrow(() -> new IOException("GraphQl schema resource 'commerce-base-schema.graphql' not found."));
+  }
+
+  @Override
+  @SuppressWarnings("deprecation")
+  public void configurePathMatch(PathMatchConfigurer matcher) {
+    matcher.setUseTrailingSlashMatch(true);
+  }
+
+  @Override
+  public void addResourceHandlers(ResourceHandlerRegistry registry) {
+    List<String> resources = new ArrayList<>(Arrays.asList("/static/**", "/docs/**"));
+    List<String> resourceLocations = new ArrayList<>();
+    if (caasServiceConfigurationProperties.isPreview()) {
+      resourceLocations.add("classpath:/static/");
+      resourceLocations.add("classpath:/META-INF/resources/");
+    }
+    registry.addResourceHandler(resources.toArray(new String[0]))
+            .addResourceLocations(resourceLocations.toArray(new String[0]));
+  }
+
+  @Bean
+  @ConditionalOnProperty("caas.logRequests")
```

Comment:
💣 always use the canonical kebab case form of properties to avoid subtle bugs. Turn this into `caas.log-requests`. If this was copied from some base headless server class, then create a jira issue to fix it there as well.

Reply:
https://coremediagmbh.atlassian.net/browse/CMS-24791